### ISSUE TITLE
suite unresponsive while reload takes a long time

### ISSUE THREAD

**Issue by matthewrmshin** [cylc/cylc#2306]
A user reported that a suite became unresponsive while it was reloading. A quick look at the suite log suggests that it spent more than 10 minutes in the [TaskPool.reload_taskdefs](https://github.com/cylc/cylc/blob/7.4.0/lib/cylc/task_pool.py#L700) method. During which, commands sent to the suite (including some kills and stops) were not being processed.

Things to consider:
* Reload should not take so long.
* If reload is going to take a while, is it possible to:
  * Feedback progress to the users via the GUI?
  * Respond to some commands, e.g. kill, stop --now?

**Comment by hjoliver** [comment]
An easy fix might be to reload groups of tasks each time through the main loop, instead of all at once.

**Comment by kinow** [comment]
Going through each issue, from the oldest to the newest (lots of interesting things!). And after reading the description of this one, I wonder if there is any chance of #2964 being related to this one @matthewrmshin ? Just in case...

**Comment by matthewrmshin** [comment]
Yes, the #2964 issue was manifested by a suite that was taking 10+ minutes to reload. This is a problem we still have to look at. It really should not take 10+ minutes to reload the task definitions.

**Comment by hjoliver** [comment]
Can we have a sanitized version of the suite in question, to analyze what's going on?

**Comment by trwhitcomb** [comment]
This is a symptom of the problem in #3255.

**Comment by hjoliver** [comment]
(This may be largely fixed now, between Tim W's validation fix #3255 and spawn-on-demand task pool reduction in Cylc 8 ... but we should still check).
